### ISSUE TITLE
Android FreeRDP: a single tap is a single left click

### ISSUE THREAD

**Issue by cergutiv** [FreeRDP/FreeRDP#9343]
**Is your feature request related to a problem? Please describe.**
Right now the issue I'm facing is that, for a single left click on the host, I have to double tap. For example, in Microsoft's RDP the user only needs to single tap. Don't know if single tapping right now is used for another feature.
For example if in the host I have to double click on something, I don't need to tap 4 times (2 for each single tap, and this doesn't work all the times and you find yourself spamming taps at the screen).
If this is the case, maybe an option to disable

**Describe the solution you'd like**
When I tap a single time, is registered on the host as a left click.

**Describe alternatives you've considered**
An option to disable the current single tap feature (don't know what is that used for).

**Additional context**
My plan is to use aFreeRDP as a better alternative to Microsoft's RDP client to use it on CK65 Android scanners at my company. Right now, Microsoft RDP client is in a bugged state, the UI is unusable for the workers and the performance is awful compared to old Windows embedded 6.5 scanners (intermec ck3).
For the tests that I have done, aFreeRDP is vastly better in our case scenario, the only drawback is to double tap for a single click, which confuses the workers.

**Comment by akallabeth** [comment]
@cergutiv the `android UI` of `FreeRDP` is currently in `maintainance` mode (meaning no new features unless someone steps up to do something)
so, we´re happy to accept pull requests but are also looking for someone who keeps an eye on the android version for bugfixes.

**Comment by tvcat** [comment]
No problem with right click here, only single tap is required.
State your afreerdp version and Android OS.

**Comment by cergutiv** [comment]
> No problem with right click here, only single tap is required. State your afreerdp version and Android OS.
Sorry I forget about it.
I'm using the aFreeRDP version from the F-Droid store, version 2.10.0
This scanners are Android 10

**Comment by tvcat** [comment]
@cergutiv
Indeed the single touch is problematic for Android version 10+ after my further testing, it work fine on Android 9.1.
The single tap need to hold a bit longer to register it. So to double tap is way harder to do now as one need to time it correctly: tap > hold > tap hold.

However the touch pointer is still working, so use that until a fix is release.

**Comment by cergutiv** [comment]
I've been trying to build the apk from the code available, so I can start working on a fix for myself at least, but I came to lot of errors and such, so I needed to modify some parts of the gradle config files to get started (I'm on Windows because it's my work's laptop). Right now, when I do gradlew build I get errors:
`> Build command failed.
  Error while executing process C:\Users\cergutiv\AppData\Local\Android\Sdk\cmake\3.18.1\bin\ninja.exe with arguments {-C C:\Users\cergutiv\StudioProjects\FreeRDP\client\Android\Studio\freeRDPCore\.cxx\Debug\44i2q3c1\arm64-v8a freerdp-android}
  ninja: Entering directory `C:\Users\cergutiv\StudioProjects\FreeRDP\client\Android\Studio\freeRDPCore\.cxx\Debug\44i2q3c1\arm64-v8a'
  [1/1] Linking C shared library C:\Users\cergutiv\StudioProjects\FreeRDP\client\Android\Studio\freeRDPCore\build\intermediates\cxx\Debug\44i2q3c1\obj\arm64-v8a\libfreerdp-android.so
  FAILED: C:/Users/cergutiv/StudioProjects/FreeRDP/client/Android/Studio/freeRDPCore/build/intermediates/cxx/Debug/44i2q3c1/obj/arm64-v8a/libfreerdp-android.so
  cmd.exe /C "cd . && C:\Users\cergutiv\AppData\Local\Android\Sdk\ndk\21.4.7075529\toolchains\llvm\prebuilt\windows-x86_64\bin\clang.exe --target=aarch64-none-linux-android23 --gcc-toolchain=C:/Users/cergutiv/AppData/Local/Android/Sdk/ndk/21.4.7075529/toolchains/llvm/prebuilt/windows-x86_64 --sysroot=C:/Users/cergutiv/AppData/Local/Android/Sdk/ndk/21.4.7075529/toolchains/llvm/prebuilt/windows-x86_64/sysroot -fPIC -g -DANDROID -fdata-sections -ffunction-sections -funwind-tables -fstack-protector-strong -no-canonical-prefixes -D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security  -O0 -fno-limit-debug-info  -Wl,--exclude-libs,libgcc.a -Wl,--exclude-libs,libgcc_real.a -Wl,--exclude-libs,libatomic.a -static-libstdc++ -Wl,--build-id -Wl,--fatal-warnings -Wl,--no-undefined -Qunused-arguments -shared -Wl,-soname,libfreerdp-android.so -o C:\Users\cergutiv\StudioProjects\FreeRDP\client\Android\Studio\freeRDPCore\build\intermediates\cxx\Debug\44i2q3c1\obj\arm64-v8a\libfreerdp-android.so CMakeFiles/freerdp-android.dir/android_event.c.o CMakeFiles/freerdp-android.dir/android_freerdp.c.o CMakeFiles/freerdp-android.dir/android_jni_utils.c.o CMakeFiles/freerdp-android.dir/android_jni_callback.c.o CMakeFiles/freerdp-android.dir/android_cliprdr.c.o  C:/Users/cergutiv/StudioProjects/FreeRDP/client/Android/Studio/freeRDPCore/src/main/jniLibs/arm64-v8a/libfreerdp3.so  C:/Users/cergutiv/StudioProjects/FreeRDP/client/Android/Studio/freeRDPCore/src/main/jniLibs/arm64-v8a/libfreerdp-client3.so  C:/Users/cergutiv/StudioProjects/FreeRDP/client/Android/Studio/freeRDPCore/src/main/jniLibs/arm64-v8a/libwinpr3.so  -llog  C:/Users/cergutiv/AppData/Local/Android/Sdk/ndk/21.4.7075529/toolchains/llvm/prebuilt/windows-x86_64/sysroot/usr/lib/aarch64-linux-android/libdl.a  -ljnigraphics  -latomic -lm && cd ."
  CMakeFiles/freerdp-android.dir/android_event.c.o: In function `android_process_event':
  C:/Users/cergutiv/StudioProjects/FreeRDP/client/Android/Studio/freeRDPCore/src/main/cpp/android_event.c:(.text.android_process_event+0xc): undefined reference to `assert'
  C:/Users/cergutiv/StudioProjects/FreeRDP/client/Android/Studio/freeRDPCore/src/main/cpp/android_event.c:(.text.android_process_event+0x10): undefined reference to `assert'
  CMakeFiles/freerdp-android.dir/android_freerdp.c.o: In function `RdpClientEntry':
  C:/Users/cergutiv/StudioProjects/FreeRDP/client/Android/Studio/freeRDPCore/src/main/cpp/android_freerdp.c:(.text.RdpClientEntry+0xc): undefined reference to `assert'
  C:/Users/cergutiv/StudioProjects/FreeRDP/client/Android/Studio/freeRDPCore/src/main/cpp/android_freerdp.c:(.text.RdpClientEntry+0x10): undefined reference to `assert'
  CMakeFiles/freerdp-android.dir/android_freerdp.c.o: In function `android_thread_func':
  C:/Users/cergutiv/StudioProjects/FreeRDP/client/Android/Studio/freeRDPCore/src/main/cpp/android_freerdp.c:(.text.android_thread_func+0xf4): undefined reference to `assert'
  CMakeFiles/freerdp-android.dir/android_freerdp.c.o:C:/Users/cergutiv/StudioProjects/FreeRDP/client/Android/Studio/freeRDPCore/src/main/cpp/android_freerdp.c:(.text.android_thread_func+0xf8): more undefined references to `assert' follow
  clang: error: linker command failed with exit code 1 (use -v to see invocation)
  ninja: build stopped: subcommand failed.`

Is this because I'm trying to build the package on Windows?

**Comment by cergutiv** [comment]
I think this is duplicated from https://github.com/FreeRDP/FreeRDP/issues/4571

**Comment by akallabeth** [comment]
@cergutiv yes, the android build scripts are `bash` and assume a `unix` (the gradle just builds the android client, not the FreeRDP libraries)

**Comment by akallabeth** [comment]
and yes, duplicate, closing.

**Comment by cergutiv** [comment]
Now that I returned from holidays I will have a look building from a Linux VM

**Comment by cergutiv** [comment]
Been trying to build, and now the error that I get is that in the file "FreeRDP/client/Android/Studio/freeRDPCore/src/main/cpp/android_freerdp.c" is including a config file called config.h
This file seems to not exist and I get an error. Is this the same file than on "FreeRDP/FreeRDP/blob/master/include/freerdp/settings.h" ?

**Comment by akallabeth** [comment]
did you run the build script for the native libs before opening gradle / android studio successfully?

**Comment by cergutiv** [comment]
Oops, actually did it before I went on AL, then commited into my fork and when coming back to work, I'm using another VM, so maybe you are right hehe.

**Comment by cergutiv** [comment]
I'm still trying to build the package. But I'm stuck on:
![imagen](https://github.com/FreeRDP/FreeRDP/assets/58826991/03626826-9278-4c7e-be87-8006c4b4cb05)
As for the solution of Google, which one of the two options should I use?  
https://developer.android.com/build/dependencies#duplicate_classes

**Comment by akallabeth** [comment]
@cergutiv all we ever do is written in [docs/README.android](https://github.com/FreeRDP/FreeRDP/blob/master/docs/README.android)
